packages/ckeditor5-ui/src/tooltipmanager.ts

```diff
@@ -259,6 +260,18 @@ export default class TooltipManager extends DomEmitterMixin() {
 		}[ position ];
 	}
 
+	/**
+	 * Handles hiding tooltips on `keydown` in DOM.
+	 *
+	 * @param evt An object containing information about the fired event.
+	 * @param domEvent The DOM event.
+	 */
+	private _onKeyDown( evt: EventInfo, { key }: KeyboardEvent ) {
+		if ( key === 'Escape' && this._currentElementWithTooltip ) {
+			this._unpinTooltip();
```

Comment:
I think we need to stop propagating this event. ATM Esc will close the underlying UI despite the tooltip being displayed. In such a case, it should take two Esc key presses to close the UI.

https://github.com/ckeditor/ckeditor5/assets/1099479/fa5ccfd1-89cc-4802-80c4-4300fc97e3d8

Reply:
Yep, `.stopPropagation` solves issue.